### ISSUE TITLE
credential_profile_name option not working

### ISSUE THREAD

**Issue by rajatprofices** [elastic/beats#13472]
I am trying to configure AWS Metric beat module for my project it works fine with secret key and access key

metricbeat.modules:
- module: aws
  period: 300s
  metricsets:
    - ec2
  access_key_id: '${AWS_ACCESS_KEY_ID:""}'
  secret_access_key: '${AWS_SECRET_ACCESS_KEY:""}'
  session_token: '${AWS_SESSION_TOKEN:""}'

but when I use credential_profile_name it doesn't work and still look for access key in the logs. Though I have properly configured credential and configure files and created the required profile 

metricbeat.modules:
- module: aws
  period: 300s
  metricsets:
    - ec2
  credential_profile_name: test-mb

**Comment by kaiyan-sheng** [comment]
@rajatprofices Thanks for reporting here. Could you try the config below please?
```
- module: aws
  period: 300s
  credential_profile_name: test-mb
  metricsets:
    - ec2
```

**Comment by rajatprofices** [comment]
Still doesn't work. I get this exception:
2019-09-04T06:29:09.714Z        ERROR   instance/beat.go:877    Exiting:
missing required field accessing '0.access_key_id'
(source:'/etc/metricbeat/modules.d/aws.yml')
Exiting: missing required field accessing '0.access_key_id'
(source:'/etc/metricbeat/modules.d/aws.yml')

this is my aws.yml

# Module: aws
# Docs:
https://www.elastic.co/guide/en/beats/metricbeat/7.3/metricbeat-module-aws.html

- module: aws
  period: 300s
  credential_profile_name: ELK
  metricsets:
    - ec2
- module: aws
  period: 60s
  credential_profile_name: ELK
  metricsets:
    - rds

and this is my AWS credentials file:

[default]
aws_access_key_id = xxxxxxxxxxxxxxxxxxxxx
aws_secret_access_key =xxxxxxxxxxxxxxxxxxxxxxxxx

[ELK]
aws_ccess_key_id = xxxxxxxxxxxxxxxxxx
aws_secret_access_key = xxxxxxxxxxxxxxxxxxxxxxxxx
role_arn = arn:aws:iam::xxxxxxxxxxxxxxxx:role/Dev_xxxx

**Comment by kaiyan-sheng** [comment]
@rajatprofices Thanks for giving it a try. First question is: what version of Metricbeat are you using? 
Second question is: where is your AWS credential file located?

**Comment by rajatprofices** [comment]
I am using 7.x and my aws credential is located at
/home/ec2-user/.aws/credential

**Comment by kaiyan-sheng** [comment]
@rajatprofices Thanks! Hmm I wonder if the credential file should be called `credentials` instead of `credential`. Based on AWS documentation, aws credential files are at `~/.aws/credentials (Linux & Mac) or %USERPROFILE%\.aws\credentials (Windows)`

**Comment by rajatprofices** [comment]
That is how it is credentials

**Comment by kaiyan-sheng** [comment]
Hmm if your aws credential file is in ~/.aws/credentials and it contains the credentials under the profile name, that should work.
```
[test-mb]
aws_access_key_id = xxxx
aws_secret_access_key = xxxx
aws_session_token = xxxx
```
(aws_session_token is optional depends on if you need it or not)

While checking the credentials file example you copied earlier, under `ELK` profile name, you have
```
[ELK]
aws_ccess_key_id
```
Is that a typo while copying or that might be why this is not working for you.

**Comment by rajatprofices** [comment]
that was a typo. It seems it doesn't even look for profile,  local access
key in aws.yml is must for it

**Comment by kaiyan-sheng** [comment]
@rajatprofices One question, you don't have `access_key_id` coexist in the config with `credential_profile_name` right?

I just pulled down 7.x and _rebuilt_ the Metricbeat binary and tested with 
```
- module: aws
  period: 300s
  credential_profile_name: ELK
  metricsets:
    - ec2
- module: aws
  period: 60s
  credential_profile_name: ELK
  metricsets:
    - rds
```
Still everything works fine on my side... Sorry I can't reproduce the problem.

**Comment by kaiyan-sheng** [comment]
I think this can be closed for now. `credential_profile_name` is tested and will be released in 7.4.0.
